OpenEdXMobile/src/main/java/org/edx/mobile/base/BaseFragment.java

```diff
@@ -33,4 +47,40 @@ public void onResume() {
      */
     protected void onRevisit() {
     }
+
+    /**
+     * This method is used when app needs to check permission(s) and if permission(s) is allowed then listener
+     * method trigger otherwise in else part show popup for requesting permission then user allow or
+     * deny that permission(s) and result return in onRequestPermissionsResult method
+     *
+     * @param permissions Permissions required to use mobile resource
+     * @param requestCode unique request code for onRequestPermissionsResult method to diffrentiate different
+     *                    permissions results
+     */
+    protected void askForPermission(String[] permissions, int requestCode) {
+        if (getActivity() != null) {
+            if (PermissionsUtil.checkPermissions(permissions[0], getActivity())) {
```

Comment:
Not the right implementation, permissions other than the one at zeroth index would get ignored.